File: ki-aim-platform/ki-aim-frontend/src/app/shared/model/file-configuration.ts
import { CsvFileConfiguration } from "./csv-file-configuration";
import { XlsxFileConfiguration } from "./xlsx-file-configuration";

export enum FileType {
	CSV = "CSV",
    XLSX = "XLSX",
}

export class FileConfiguration {
    constructor(
        public fileType: FileType | null,
        public csvFileConfiguration: CsvFileConfiguration,
        public xlsxFileConfiguration: XlsxFileConfiguration,
    ) {}
}

File: cinnamon-platform/src/main/java/de/kiaim/platform/model/file/FileConfiguration.java
package de.kiaim.platform.model.file;

import io.swagger.v3.oas.annotations.media.Schema;
import jakarta.validation.Valid;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Schema(description = "Metadata describing the type of the uploaded data.")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class FileConfiguration {

	@Schema(description = "Type of the file.", example = "CSV")
	@NotNull(message = "File type must be present")
	private FileType fileType;

	@Schema(description = "Configurations specific for CSV files.")
	@NotNull(message = "CSV file configuration must be present")
	@Valid
	private CsvFileConfiguration csvFileConfiguration;

	@Schema(description = "Configuration specific for XLSX files")
	private XlsxFileConfiguration xlsxFileConfiguration;
}
